### ISSUE TITLE
crash/unhandled exception GMLAS

### ISSUE THREAD

**Issue by vautieri** [OSGeo/gdal#3566]
options = CSLSetNameValue( options, "EXPOSE_METADATA_LAYERS", "NO" );
   options = CSLSetNameValue( options, "SCHEMA_FULL_CHECKING", "NO" );
   options = CSLSetNameValue( options, "VALIDATE", "NO" );
   options = CSLSetNameValue( options, "FAIL_IF_VALIDATION_ERROR", "NO" );
   options = CSLSetNameValue( options, "REMOVE_UNUSED_LAYERS", "NO" );
   options = CSLSetNameValue( options, "REMOVE_UNUSED_FIELDS", "NO" );
   options = CSLSetNameValue( options, "HANDLE_MULTIPLE_IMPORTS", "YES" );

If you use the above, which is not the same as the config file defaults, one will crash.  If you change VALIDATE to yes or REMOVE_UNUSED_LAYERS to use, you can get around the crash for those needing a quick fix without a recompile.

in file ogrgmlasdatasource.cpp, if you uncomment this line, you will not crash.  The code is hard to follow with the VSIFILE (what appears to be the issue), I think something in RunFirstPassIfNeeded( nullptr, nullptr, nullptr ); initializes something properly.

    //if( m_bValidate || m_bRemoveUnusedLayers ||
    //    (m_bFoundSWE && (m_oConf.m_bSWEProcessDataRecord ||
    //                     m_oConf.m_bSWEProcessDataArray)) )

You crash on "VSIFSeekL(fp, 0, SEEK_SET);" at the below:
GMLASBinInputStream::GMLASBinInputStream(VSILFILE* fp)
{
    m_fp = fp;
    VSIFSeekL(fp, 0, SEEK_SET);
}

fp is NOT NULL.  I did go through and set a couple places where fp isn't set to null after its "push" for reuse.  ..didn't work.  

This is version gdal-3.2.1.

We came from GDAL version 2.4, and we do not crash on that older version of gdal.  We do open the file multiple times (as on our background threads), and we are properly opening/shared on each thread (we have been doing i this way for years).   I spent a couple hours trying to figure out the code paths and am at a bit of a loss.   I consider what I did above as a hack, but it at least keeps the overhead down so you don't have to validate all the time (we need this to be quick processing)

**Comment by rouault** [comment]
@vautieri Could you give the above suggested feedback ?

**Comment by vautieri** [comment]
Will do... we are just swamped getting this project completed(all works
long hours)

**Comment by rouault** [comment]
Closing due to lack of feedback
